src/Database/Adapter/MariaDB.php

```diff
@@ -309,18 +309,34 @@ public function createAttribute(string $collection, string $id, string $type, in
      * @param bool $signed
      * @param bool $array
      * @param string $newKey
+     * @param int $newSize
      * @return bool
      * @throws Exception
      * @throws PDOException
      */
-    public function updateAttribute(string $collection, string $id, string $type, int $size, bool $signed = true, bool $array = false, string $newKey = null): bool
+    public function updateAttribute(string $collection, string $id, string $type, int $size, bool $signed = true, bool $array = false, string $newKey = null, int $newSize = null): bool
     {
         $name = $this->filter($collection);
         $id = $this->filter($id);
-        $type = $this->getSQLType($type, $size, $signed, $array);
+        $newKey = empty($newKey) ? null : $this->filter($newKey);
+        $type = $this->getSQLType($type, $newSize ?? $size, $signed, $array);
+
+        if (!empty($newSize) && $size > $newSize) {
+            $sql = "
+                ALTER TABLE {$this->getSQLTable($collection)} ADD COLUMN `new_{$id}` {$type};
+                UPDATE {$this->getSQLTable($collection)} SET `new_{$id}` = LEFT(`{$id}`, {$newSize});
+                ALTER TABLE {$this->getSQLTable($collection)} DROP COLUMN `{$id}`;
+                ALTER TABLE {$this->getSQLTable($collection)} RENAME COLUMN `new_{$id}` TO `". ($newKey ?? $id) ."`;
```

Comment:
Let's try a simpler approach, that while locking tables, will reduce edge cases, and ensure data consistency

```suggestion
                LOCK TABLES {$this->getSQLTable($collection)} WRITE;
                UPDATE {$this->getSQLTable($collection)} SET `{$id}` = LEFT(`{$id}`, {$newSize});."`;
                UNLOCK TABLES;
```